src/Tree/Strategy/ORM/Closure.php

```diff
@@ -329,7 +347,16 @@ public function processPostPersist($em, $entity, AdapterInterface $ea)
                 $levelProp->setValue($node, 1);
             }
 
-            foreach ($entries as $closure) {
+            foreach ($entries as $key => $closure) {
+                $ancestorColumnName = $entriesMeta[$key]['ancestorColumnName'];
```

Comment:
IMO, we should merge the `$entries` and the `$entriesMeta` arrays into a single one.
Moreover, if possible, I think we could use a solution based on generators to reduce the array's usage.

Reply:
Can you say more about what you mean by using generators? Do you mean that I should repeat the code in lines 285-297 to get the required names (i.e., by pulling these lines out into a separate method)? That would be a better solution than what I have here, but perhaps you meant something else?

Reply:
I think I was making it more complicated than it needs to be. All these properties stay the same through the rest of this loop, so I don't need to store the property names for each separate entry. Have a look at my latest commit.

Reply:
I agree, thank you.